servers/castor/edna.nix

```diff
@@ -0,0 +1,121 @@
+{pkgs, lib, config, ...}:
+  let
+    inherit (builtins) toString typeOf toJSON;
+    inherit (lib) collect concatStringsSep;
+    inherit (pkgs) writeText;
+
+    dns_names = with config.networking;
+      concatStringsSep "," [ "${hostName}.${domain}" "staging.edna.serokell.team" ];
+
+    profile = "/nix/var/nix/profiles/per-user/deploy/edna-docker";
+
+    ports = {
+      backend = {
+        api = 9000;
+      };
+      frontend = {
+        http = 80;
+        https = 443;
+      };
+      prometheus = {
+        node = 9100;
+      };
+    };
+
+    pginit = writeText "init.sql"
+    ''
+      CREATE DATABASE edna;
+    '';
+  in
+  {
+    networking.firewall.allowedTCPPorts = collect (a: (typeOf a) == "int") { inherit (ports) frontend prometheus; };
+
+    virtualisation.docker = {
+      enable = true;
+      logLevel = "warn";
+      storageDriver = "overlay2";
+      volumes = [ "pgdata" ];
+      networks.edna = {};
+    };
+
+    systemd.services.docker-frontend.serviceConfig.RestartSec = "5m";
+
+    virtualisation.oci-containers.containers = let
+      commonOptions = {
+        extraOptions = [
+          # Put all nodes on the same private network
+          "--network=edna"
+
+          # PostgreSQL during initdb
+          "--shm-size=256MB"
+        ];
+      };
+
+    in
+    {
+      frontend = {
+        image = "ghcr.io/serokell/edna-backend";
+        imageFile = "${profile}/ghcr.io-serokell-edna-frontend.tar.gz";
+        dependsOn = [ "backend" ];
+
+        volumes = [
+          "/root/caddy:/.caddy"
+        ];
+
+        ports = with ports.frontend; [
+          "${toString http}:${toString http}"
+          "${toString https}:${toString https}"
+        ];
+
+        environment = {
+          API_HOST = "backend";
+          API_PORT = toString ports.backend.api;
+          DNS_DOMAIN = dns_names;
```

Comment:
Who is the consumer of these 3 env variables?

(probably a stupid question, I just didn't deal with frontend before)

Reply:
They are used in the Caddyfile: https://github.com/serokell/edna/blob/mkaito/edna14-oci-images/docker.nix#L38